compiler/src/iree/compiler/Codegen/Common/GPU/GPUNestedLayoutDistributionPatterns.cpp

```diff
@@ -305,7 +305,11 @@ struct DistributeBroadcast final : OpDistributionPattern<vector::BroadcastOp> {
     auto vectorType = VectorType::get(distShape, elementType);
 
     VectorValue srcVector = dyn_cast<VectorValue>(broadcastOp.getSource());
-    if (!srcVector) {
+    // Types such as vector<f32> can return a valid pointer. An additional
+    // rank check is added to ensure that the type is indeed a vector
+    // value.
+    bool isSrcVector = (srcVector) && (isVector(srcVector));
+    if (!isSrcVector) {
```

Comment:
This is the only use of `isSrcVector`, so it makes sense to inline it
```suggestion
    if (!srcVector || !isVector(srcVector)) {
```